airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -398,27 +399,47 @@ def read_records(
         seconds_waited = 0
         try:
             report_id = self._create_report(sync_mode, cursor_field, stream_slice, stream_state)["reportId"]
-        except DefaultBackoffException as e:
-            logger.warning(f"The report for stream '{self.name}' was cancelled due to several failed retry attempts. {e}")
-            return []
         except requests.exceptions.HTTPError as e:
+            errors = " ".join([er.get("message", "") for er in e.response.json().get("errors", [])])
+            if e.response.status_code == requests.codes.BAD_REQUEST:
+                invalid_report_names = list(
+                    map(
+                        lambda error: error.get("message").replace("Invalid Report Type ", ""),
+                        filter(lambda error: "Invalid Report Type " in error.get("message"), e.response.json().get("errors", [])),
+                    )
+                )
+                if invalid_report_names:
+                    raise AirbyteTracedException(
+                        failure_type=FailureType.config_error,
+                        message=f"Report {invalid_report_names} does not exist. Please update the report options in your config to match only existing reports.",
+                        internal_message=f"Errors received from the API were: {errors}",
+                    )
             if e.response.status_code == requests.codes.FORBIDDEN:
-                logger.warning(
-                    f"The endpoint {e.response.url} returned {e.response.status_code}: {e.response.reason}. "
+                raise AirbyteTracedException(
+                    failure_type=FailureType.config_error,
+                    message=f"The endpoint {e.response.url} returned {e.response.status_code}: {e.response.reason}. "
                     "This is most likely due to insufficient permissions on the credentials in use. "
-                    "Try to grant required permissions/scopes or re-authenticate."
+                    "Try to grant required permissions/scopes or re-authenticate.",
+                    internal_message=f"Errors received from the API were: {errors}",
+                )
+            if e.response.status_code == requests.codes.TOO_MANY_REQUESTS:
+                raise AirbyteTracedException(
+                    failure_type=FailureType.transient_error,
+                    message=f"Too many requests on resource {e.response.url}. Please retry later",
+                    internal_message=f"Errors received from the API were: {errors}",
                 )
-                return []
 
-            errors = " ".join([er.get("message", "") for er in e.response.json().get("errors", [])])
             if "does not support account ID of type class com.amazon.partner.account.id.VendorGroupId." in errors:
-                logger.warning(
-                    f"The endpoint {e.response.url} returned {e.response.status_code}: {errors}. "
+                raise AirbyteTracedException(
```

Comment:
I'm fine with updating it as config error. Now, with updated error handling, all errors use AirbyteTracedException instead of logging. Thanks. 
JFYI, you can also add report_id to error message, so airbyte users or devs can share this id with amazon seller partner support and they can provide more detailed info regarding failure for specific report id.

Reply:
generally we can use report id where it's possible, amazon support asked it when I contact them regarding other issue with fatal statuses, so we could provide more info regarding errors when we will need to contact support